### ISSUE TITLE
[FPEnv][AArch64] Issues around -fexperimental-strict-floating-point

### ISSUE THREAD

**Issue by sfc-gh-sgiesecke** [llvm/llvm-project#54975]
When compiling:
```
#include <fenv.h>

int main() {
    fesetround(FE_TOWARDZERO);
}
```
with
```
clang -frounding-math -fexperimental-strict-floating-point test.c -c
```
this yields two warnings:
```
clang-14: warning: argument unused during compilation: '-fexperimental-strict-floating-point' [-Wunused-command-line-argument]
warning: overriding currently unsupported rounding mode on this target [-Wunsupported-floating-point-opt]
1 warning generated.
```
(not sure why this says "1 warning", but it does).

Both warnings are unexpected:
* Re `-Wunused-command-line-argument`: `-fexperimental-strict-floating-point` is a documented command line option. Interestingly, this warning goes away when removing `-c` and create a binary. Still, this seems wrong as this doesn't have anything to do with linking.
* Re `-Wunsupported-floating-point-opt`: My assumption was that `-fexperimental-strict-floating-point` removes this warning, and actually enables that support.

More generally, what's the current state of the `-frounding-math` implementation on `AArch64`? Is this going to be non-experimental in the next release?
